include/sys/abd.h

```diff
@@ -174,13 +133,29 @@ abd_zero(abd_t *abd, size_t size)
 	abd_zero_off(abd, 0, size);
 }
 
+/*
+ * ABD type check functions
+ */
+boolean_t abd_is_linear(abd_t *);
+boolean_t abd_is_multi(abd_t *);
+boolean_t abd_is_linear_page(abd_t *);
+
 /*
  * Module lifecycle
+ * Defined in each specific OS's abd.c
  */
 
 void abd_init(void);
 void abd_fini(void);
 
+/*
+ * Linux ABD bio functions
+ */
```

Comment:
should these be "ifdef LINUX"?

Reply:
This can be changed to just be #ifdef LINUX. Simple change and would isolate this just to Linux builds.

Reply:
I wound up changing this to be #if defined(__linux__) with the _KERNEL check as well. I also wound creating a separate commit to just contain moving the ABD code into separate OS files for Linux and FreeBSD. This should make seeing only the Gang ABD changes much easier.